website/src/components/Messages/MessageTableEntry.tsx

```diff
@@ -155,16 +158,47 @@ const MessageActions = ({
   onReport: () => void;
   message: Message;
 }) => {
+  const toast = useToast();
   const { t } = useTranslation(["message", "common"]);
-  const { trigger } = useSWRMutation(`/api/admin/delete_message/${message.id}`, del);
-  const { data } = useSession() || {};
-  const role = data?.user?.role;
+  const id = message.id;
+  const displayId = id.slice(0, CHAR_COUNT) + "..." + id.slice(-CHAR_COUNT);
+  const { trigger: deleteMessage } = useSWRMutation(`/api/admin/delete_message/${message.id}`, del);
+
+  const { trigger: stopTree } = useSWRMutation(`/api/admin/stop_tree/${message.id}`, put, {
+    onSuccess: () => {
+      toast({
+        title: t("common:success"),
+        description: t("tree_stopped", { id: displayId }),
+        status: "success",
+        duration: 5000,
+        isClosable: true,
+      });
+    },
+  });
+
+  const { data: session } = useSession();
+  const isAdmin = session?.user?.role === "admin";
```

Comment:
This logic is quite often, maybe we should extract it to useHasRole(role: Role) hook?

Reply:
I think this is probably out of scope for this PR and better refactored in a separate pull request.

Reply:
I agree we should clean this up into a hook.  In a follow up PR works for me.

Reply:
I welcome you to try that. `setValue` does not work as expected and you need to use an effect to make it work because we don't have an on change function. I added this feature because there's no easy way to know what message you are deleting or halting on the UI.